modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -310,94 +276,58 @@ def get_files_info(ip, _rport, shares, info)
     end
   end
 
-  def rport
-    @rport || datastore['RPORT']
-  end
-
-  # Overrides the one in smb.rb
-  def smb_direct
-    @smb_redirect || datastore['SMBDirect']
-  end
-
   def run_host(ip)
-    @rport = datastore['RPORT']
-    @smb_redirect = datastore['SMBDirect']
-    @srvsvc = datastore['USE_SRVSVC_ONLY']
     shares = []
+    begin
+      print_status 'Starting module'
+      connect(versions: [1, 2, 3])
+      smb_login
+      shares = simple.client.net_share_enum_all(ip)
+      os_info = get_os_info(ip)
+      print_status(os_info) if os_info
+
+      if shares.empty?
+        print_status('No shares collected')
+      else
+        shares.each do |share|
+          print_good("#{share[:name]} - (#{share[:type]}) #{share[:comment]}")
+        end
 
-    [[139, false], [445, true]].each do |info|
```

Comment:
For consistency with the other modules we should leave this 139/445 mechanism in place